### PR TITLE
Configure and add example for `PrincipalAssociation.ram` and `ResourceShareAccepter.ram`

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

In this PR: `PrincipalAssociation` and  `ResourceShareAccepter` resources have been configured and added examples.

Fixes: https://github.com/upbound/provider-aws/issues/253

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

- `aws_ram_principal_association`: Uptest(https://github.com/upbound/provider-aws/actions/runs/5811127760)
- `aws_ram_resource_share_accepter`: Manually, Uptest(https://github.com/upbound/provider-aws/actions/runs/5831983029)
```
NAME                                                       READY   SYNCED   EXTERNAL-NAME
                                      AGE
resourceshareaccepter.ram.aws.upbound.io/receiver-accept   True    True     arn:aws:ram:us-west-1:243569053065:resource-share/d
bdb11d5-6d69-43bf-90e8-ce0db2ff9dab   25m

NAME                                            READY   SYNCED   EXTERNAL-NAME
                           AGE
resourceshare.ram.aws.upbound.io/sender-share   True    True     arn:aws:ram:us-west-1:243569053065:resource-share/dbdb11d5-6d6
9-43bf-90e8-ce0db2ff9dab   66m

NAME                                                    READY   SYNCED   EXTERNAL-NAME
                                                AGE
principalassociation.ram.aws.upbound.io/sender-invite   True    True     arn:aws:ram:us-west-1:243569053065:resource-share/dbdb
11d5-6d69-43bf-90e8-ce0db2ff9dab,153891904029   66m
```

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/principalassociation.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/directconnect/hostedprivatevirtualinterfaceaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ram/resourceshareaccepter.yaml"

File: examples/ram/resourceshareaccepter.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+metadata:
+  annotations:
+    meta.upbound.io/example-id: ram/v1beta1/resourceshareaccepter
+    uptest.upbound.io/pre-delete-hook: testhooks/pre-delete-principal.sh
+  labels:
+    testing.upbound.io/example-name: example
+  name: sender-invite
+spec:
+  providerConfigRef:
+    name: peer
```
Comment (ulucinar) [inline examples/ram/resourceshareaccepter.yaml:28]: Thank you @turkenf for your dedication in improving our test automation and going the extra mile to keep our resources uptestable, even if they are cross-account resources.

**Review by ulucinar — APPROVED** [review]
Thank you @turkenf, lgtm.

**Pull request merged**